### PR TITLE
Fix HTTP connection error for long running transfers

### PR DESCRIPTION
**Pull request by sarahwooders**
Implements a few bug fixes causing errors for large transfers: 
* A basic backpressure mechanism, so that if the queues on a gateway are full, the `chunk_requests` POST request will return how many chunks were added and the current queue size, informing the HTTP client making the request to send the remaining chunks (those not added) to a different gateway or to wait and try again. With this change, I was able to transfer 1TB. 
* This also reduces the *total* number of HTTP connections per gateway to be 64, as opposed to 32 per destination, which seems to have been causing issues. 
* Empty chunks are allowed, since object stores can have empty folders which we still want transferred 

There are still issues for SSH connections for long running transfers, and listing files can take an extremely long time on the client (#841), so these issues need to be fixed to for very large transfers.

### CODE REVIEW

File: skyplane/api/tracker.py [hunk 1]
```diff
@@ -146,22 +146,25 @@ def run(self):
...
                 )
+
         except Exception as e:
             UsageClient.log_exception(
                 "dispatch job",
                 e,
                 args,
                 self.dataplane.topology.src_region_tag,
-                self.dataplane.topology.dest_region_tags,
+                self.dataplane.topology.dest_region_tags[0],  # TODO: support multiple destinations
```
Comment (simon-mo) [inline skyplane/api/tracker.py:167]:
```suggestion
                *self.dataplane.topology.dest_region_tags,  
```
You can just "spread" this list as arguments

File: skyplane/api/transfer_job.py [hunk 2]
```diff
@@ -630,24 +633,30 @@ def dispatch(
...
+
+                # TODO: make async
+                reply = self.http_pool.request(
+                    "POST",
+                    f"{server.gateway_api_url}/api/v1/chunk_requests",
+                    body=json.dumps([chunk.as_dict() for chunk in chunk_batch[n_added:]]).encode("utf-8"),
+                    headers={"Content-Type": "application/json"},
+                )
+                reply_json = json.loads(reply.data.decode("utf-8"))
+                print(server, min_idx, "added", n_added, len(chunk_batch), reply_json)
```
Comment (simon-mo) [inline skyplane/api/transfer_job.py:651]:
```suggestion
```

Comment (simon-mo) [reply skyplane/api/transfer_job.py:651]: log debug messages?

File: skyplane/gateway/operators/gateway_operator.py [hunk 3]
```diff
@@ -78,11 +78,14 @@ def worker_loop(self, worker_id: int, *args):
         self.worker_id = worker_id
         while not self.exit_flags[worker_id].is_set() and not self.error_event.is_set():
             try:
+                # print(f"[{self.handle}:{self.worker_id}] Waiting for chunk, queue size {self.input_queue.size()}")
```
Comment (simon-mo) [inline skyplane/gateway/operators/gateway_operator.py:81]: clean up this file a bit?

File: skyplane/gateway/operators/gateway_operator.py [hunk 4]
```diff
@@ -475,7 +497,14 @@ def process(self, chunk_req: ChunkRequest, **args):
         # while self.chunk_store.remaining_bytes() < chunk_req.chunk.chunk_length_bytes * self.n_processes:
         #    time.sleep(0.1)
 
-        assert chunk_req.chunk.chunk_length_bytes > 0, f"Cannot have size 0 chunk {chunk_req.chunk}"
+        # assert chunk_req.chunk.chunk_length_bytes > 0, f"Cannot have size 0 chunk {chunk_req.chunk}" # actually ok
+
+        if chunk_req.chunk.chunk_length_bytes == 0:
+            # nothing to do
+            # create empty file
+            open(fpath, "a").close()
```
Comment (simon-mo) [inline skyplane/gateway/operators/gateway_operator.py:505]: Path('path/to/file.txt').touch()

**Review by simon-mo — APPROVED** [review]

File: skyplane/api/tracker.py (hunk 1 above)
Comment (sarahwooders) [reply skyplane/api/tracker.py:167]: what does this mean?

**Pull request merged**